app/code/Magento/LayeredNavigation/Block/Navigation.php

```diff
@@ -107,7 +107,7 @@ public function getFilters()
      */
     public function canShowBlock()
     {
-        return $this->visibilityFlag->isEnabled($this->getLayer(), $this->getFilters());
+        return $this->getLayer()->getCurrentCategory()->getDisplayMode() !== \Magento\Catalog\Model\Category::DM_PAGE;
```

Comment:
@priti2jcommerce I think both conditions should be checked here: display mode and visibility flag